modules/caption-impl/src/main/java/org/opencastproject/caption/converters/WebVttCaptionConverter.java

```diff
@@ -107,7 +124,41 @@ public String getExtension() {
 
   @Override
   public Type getElementType() {
-    return MediaPackageElement.Type.Attachment;
+    return mediapackageElementType;
+  }
+
+  /**
+   * Converts the configured property 'mediapackage-element-type' into the corresponding enum 'MediapackageElement.Type'
+   *
+   * @param cc ComponentContext
+   * @return the configured mediapackage element type
+   */
+  private MediaPackageElement.Type getConfiguredMediapackageElementType(ComponentContext cc) {
+    Option<String> mediapackageElementTypeOption = OsgiUtil.getOptCfg(
+        cc.getProperties(), MEDIAPACKAGE_ELEMENT_TYPE_CONFIG_KEY);
+    if (mediapackageElementTypeOption.isNone()) {
+      return DEFAULT_MEDIAPACKAGE_ELEMENT_TYPE; // returning default if config isn't set
+    }
+    return convertStringToEnum(mediapackageElementTypeOption.get());
+  }
+
+  /**
+   * Helps to convert a configured mediaPackageElementType.
+   * Converts from a string to the corresponding enum.
+   *
+   * @param mediaPackageElementType The string that will be converted to the enum.
+   * @return The resulting enum after the conversion.
+   */
+  private Type convertStringToEnum(String mediaPackageElementType) {
+    if (EnumUtils.isValidEnumIgnoreCase(MediaPackageElement.Type.class, mediaPackageElementType)) {
+      return EnumUtils.getEnumIgnoreCase(MediaPackageElement.Type.class, mediaPackageElementType);
+    }
+    // Conversion didn't work. Print an error and throw an exception
+    String additionalErrorInfo = String.format("Please check the config file '%s'.", this.getClass().getName());
+    String errorMessage = String.format("Couldn't convert configuration '%s'='%s' into enum. " + additionalErrorInfo,
+        MEDIAPACKAGE_ELEMENT_TYPE_CONFIG_KEY, mediaPackageElementType);
+    logger.error(errorMessage);
+    throw new IllegalArgumentException(errorMessage);
```

Comment:
Why log this if you throw this as an error, which will then be logged as an error?
Just throwing the error should be sufficient and should also make the `format()` unnecessary.

Reply:
I removed the logging line and thinned out the error message.